scripts/localization.swift

```diff
@@ -0,0 +1,349 @@
+		        if numberString == "0" {
+		            actualComponent = (component as NSString).substring(from: match.range.length) as String?
+		            keyForComponent = "zero"
+		        }
+        
+		    } else {
+		    	if keyIndex < keys.count {
+		    		keyForComponent = keys[keyIndex]
+					keyIndex += 1
+		    	}
+		    }
+			
+			guard let keyToInsert = keyForComponent, let componentToInsert = actualComponent else {
+				continue
+			}
+			
+			keyDictionary[keyToInsert] = self.replacingCharacters(in: range, with: componentToInsert).replacingOccurrences(of: token, with: newToken)
+			
+		}
+        
+        let key = "v0"
+        mutableDictionary[key] = keyDictionary
+        let replacement = "%#@\(key)@"
+        mutableDictionary["NSStringLocalizedFormatKey"] = replacement
+        return mutableDictionary
+    }
+    
+    public func replacingMatches(fromRegex regex: NSRegularExpression, withFormat format: String) -> String {
+        let nativeLocalization = NSMutableString(string: self)
+        var offset = 0
+        let fullRange = NSRange(location: 0, length: nativeLocalization.length)
+        regex.enumerateMatches(in: self, options: [], range: fullRange) { (result, flags, stop) in
+            guard let result = result else {
+                return
+            }
+            let token = regex.replacementString(for: result, in: nativeLocalization as String, offset: offset, template: "$1")
+            let replacement = String(format: format, token)
+            let replacementRange = NSRange(location: result.range.location + offset, length: result.range.length)
+            nativeLocalization.replaceCharacters(in: replacementRange, with: replacement)
+            offset += (replacement as NSString).length - result.range.length
+        }
+        return nativeLocalization as String
+    }
+    
+    var iOSNativeLocalization: String {
+        guard let tokenRegex = twnTokenRegex else {
+            return ""
+        }
+        return self.replacingMatches(fromRegex: tokenRegex, withFormat: "%%%@$@")
+    }
+    
+    var twnNativeLocalization: String {
+        guard let tokenRegex = iOSTokenRegex else {
+            return ""
+        }
+        return self.replacingMatches(fromRegex: tokenRegex, withFormat: "$%@")
+    }
+}
+
+func writeStrings(fromDictionary dictionary: [String: String], toFile: String, escaped: Bool) throws {
+    var output = ""
+    let sortedDictionary = dictionary.sorted(by: { (kv1, kv2) -> Bool in
+        return kv1.key < kv2.key
+    })
+    for (key, value) in sortedDictionary {
+        output.append("\"\(key)\" = \"\(escaped ? value.escapedString : value)\";\n")
+    }
+    
+    try output.write(toFile: toFile, atomically: true, encoding: .utf8)
+}
+
+
+func exportLocalizationsFromSourceCode() {
+	let basePath = "Wikipedia/iOS Native Localizations/Base.lproj/Localizable.strings"
```

Comment:
This file is kind of a mess, but here's where we take the `Base.lproj/Localizable.strings` generated by `xcrun extractLocStrings` and generate the `qqq.lproj/Localizable.strings` (with translation comments) and `en.lproj/Localizable.strings` (with english translations) that TWN looks for. Note that all TWN localizations are in `Wikipedia/Localizations` all iOS native localizations that we use in the app are in `Wikipedia/iOS Native Localizations`